src/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -0,0 +1,251 @@
+   * @param handler {Object} Reference to the Handler Class Constructor
+   * @param el {HTMLElement} HTML element to which the chart will be appended
+   * @param chartData {Object} Elasticsearch query results for this specific chart
+   */
+  class ColumnChart extends PointSeries {
+    constructor(handler, chartEl, chartData, seriesConfigArgs) {
+      super(handler, chartEl, chartData, seriesConfigArgs);
+      this.seriesConfig = _.defaults(seriesConfigArgs || {}, defaults);
+    }
+
+    addBars(svg, data) {
+      const self = this;
+      const color = this.handler.data.getColorFunc();
+      const tooltip = this.baseChart.tooltip;
+      const isTooltip = this.handler.visConfig.get('tooltip.show');
+
+      const layer = svg.append('g')
+      .attr('class', function (d, i) {
+        return 'series ' + i;
+      });
+
+      const bars = layer.selectAll('rect')
+      .data(data.values);
+
+      bars
+      .exit()
+      .remove();
+
+      bars
+      .enter()
+      .append('rect')
+      .attr('data-label', data.label)
+      .attr('fill', () => {
+        return color(data.label);
+      });
+
+      self.updateBars(bars);
+
+      // Add tooltip
+      if (isTooltip) {
+        bars.call(tooltip.render());
+      }
+
+      return bars;
+    };
+
+    /**
+     * Determines whether bars are grouped or stacked and updates the D3
+     * selection
+     *
+     * @method updateBars
+     * @param bars {D3.UpdateSelection} SVG with rect added
+     * @returns {D3.UpdateSelection}
+     */
+    updateBars(bars) {
+      if (this.seriesConfig.mode === 'stacked') {
+        return this.addStackedBars(bars);
+      }
+      return this.addGroupedBars(bars);
+
+    };
+
+    /**
+     * Adds stacked bars to column chart visualization
+     *
+     * @method addStackedBars
+     * @param bars {D3.UpdateSelection} SVG with rect added
+     * @returns {D3.UpdateSelection}
+     */
+    addStackedBars(bars) {
+      const xScale = this.getCategoryAxis().getScale();
+      const yScale = this.getValueAxis().getScale();
+      const isHorizontal = this.getCategoryAxis().axisConfig.isHorizontal();
+      const height = yScale.range()[0];
+      const yMin = yScale.domain()[0];
+
+      let barWidth;
+      if (this.getCategoryAxis().axisConfig.isTimeDomain()) {
+        const { min, interval } = this.handler.data.get('ordered');
+        const start = min;
+        const end = moment(min).add(interval).valueOf();
+
+        barWidth = xScale(end) - xScale(start);
+        if (!isHorizontal) barWidth *= -1;
+        barWidth = barWidth - Math.min(barWidth * 0.25, 15);
+      }
+
+      function x(d) {
+        return xScale(d.x);
+      }
+
+      function y(d) {
+        return yScale(d.y0 + d.y);
+      }
+
+      function widthFunc() {
+        return barWidth || xScale.rangeBand();
+      }
+
+      function heightFunc(d) {
+        if (d.y < 0) {
+          return Math.abs(yScale(d.y0 + d.y) - yScale(d.y0));
+        }
+        // todo:
```

Comment:
Should we create an issue and refer to the issue URL instead of an inline comment? I saw Tim doing that and thought it seemed like a good way to make sure these little TODOs don't get lost.